DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/FlagRush_RpcEvents.Script.txt

```diff
@@ -0,0 +1,280 @@
+
+	XmlRpc::RegisterCallback(C_XMLRPC_MatchStart, "Called when event occurs");
+	XmlRpc::RegisterCallback(C_XMLRPC_MatchEnd, "Called when event occurs");
+	XmlRpc::RegisterCallback(C_XMLRPC_MapStart, "Called when event occurs");
+	XmlRpc::RegisterCallback(C_XMLRPC_MapEnd, "Called when event occurs");
+	XmlRpc::RegisterCallback(C_XMLRPC_RoundStart, "Called when event occurs");
+	XmlRpc::RegisterCallback(C_XMLRPC_RoundEnd, "Called when event occurs");
+}
+
+Void SetMatchStart() {
+	G_MatchStartTime = Now;
+}
+
+Integer GetMatchTime() {
+	return Now - G_MatchStartTime;
+}
+
+K_Team GetTeamFromLogin(Text Login) {
+	declare CSmPlayer Player <=> GetPlayer(Login);
+	declare CTeam Team;
+	if (Player == Null) return K_Team{};
+	if (Player.CurrentClan > 0) Team <=> Teams[Player.CurrentClan - 1];
+	if (Team == Null) return K_Team{};
+	return K_Team{Num = Player.CurrentClan, Name = Team.Name };
+}
+
+K_RpcScore GetMatchScore(CSmScore Score) {
+	return K_RpcScore{
+		MatchPoints = Scores::GetPlayerMatchPoints(Score),
+		FlagsScored = Telemetry::GetPlayerMatchInteger(Score, FlagRush_Common::C_Telemetry_Key_FlagsScored),
+		FlagsStealed = Telemetry::GetPlayerMatchInteger(Score, FlagRush_Common::C_Telemetry_Key_FlagsStolen),
+		FlagsAssists = Telemetry::GetPlayerMatchInteger(Score, FlagRush_Common::C_Telemerty_Key_FlagScoreAssists)
+	};
+}
+
+K_RpcScore GetMapScore(CSmScore Score) {
+	return K_RpcScore{
+		MatchPoints = Scores::GetPlayerMatchPoints(Score),
+		FlagsScored = Telemetry::GetPlayerMapInteger(Score, FlagRush_Common::C_Telemetry_Key_FlagsScored),
+		FlagsStealed = Telemetry::GetPlayerMapInteger(Score, FlagRush_Common::C_Telemetry_Key_FlagsStolen),
+		FlagsAssists = Telemetry::GetPlayerMapInteger(Score, FlagRush_Common::C_Telemerty_Key_FlagScoreAssists)
+	};
+}
+
+
+K_RpcScore GetRoundScore(CSmScore Score) {
+	return K_RpcScore{
+		MatchPoints = Scores::GetPlayerMatchPoints(Score),
+		FlagsScored = Telemetry::GetPlayerRoundInteger(Score, FlagRush_Common::C_Telemetry_Key_FlagsScored),
+		FlagsStealed = Telemetry::GetPlayerRoundInteger(Score, FlagRush_Common::C_Telemetry_Key_FlagsStolen),
+		FlagsAssists = Telemetry::GetPlayerRoundInteger(Score, FlagRush_Common::C_Telemerty_Key_FlagScoreAssists)
+	};
+}
+
+K_RpcPlayer GetRpcPlayer(CSmPlayer Player) {
+	if (Player == Null) return K_RpcPlayer{};
+	declare CTeam Team;
+	if (Player.CurrentClan > 0) Team <=> Teams[Player.CurrentClan - 1];
+	if (Team == Null) return K_RpcPlayer{Login = Player.User.Login, Name = Player.User.Name,
+		Team = Player.CurrentClan, Position = Player.Position};
+	return K_RpcPlayer{Login = Player.User.Login, Name = Player.User.Name,
+		Team = Player.CurrentClan, TeamName = Team.Name, Position = Player.Position
+	};
+}
+
+K_RpcPlayerScore GetRpcPlayerFromScore(CSmScore Score, K_Team _Team) {
+	if (Score == Null || Score.User == Null) return K_RpcPlayerScore{};
+	return K_RpcPlayerScore{Login = Score.User.Login, Name = Score.User.Name,
+		Team = _Team.Num, TeamName = _Team.Name,
```

Comment:
If K_Team is gonna be changed, then we can get the `_Team.Num` with `Teams.keyof(Team)`, where `Team` is of type `CTeam`.